### PR TITLE
Update gcloud-datastore to use local development emulator

### PR DESCRIPTION
**Pull request by nickfirespotter**
Check for the DATASTORE_EMULATOR_HOST env variable and if it exists assume working in dev mode. Then update the API_ROOT to point to the emulator as well as not requiring Authorization header or the need to obtain an auth Token.

### CODE REVIEW

File: datastore/gcloud/aio/datastore/datastore.py [hunk 1]
```diff
@@ -43,13 +52,21 @@ def __init__(self, project: Optional[str] = None,
...
+            self.token = None
+        else:
+            self.token = token or Token(service_file=service_file, session=session,
+                                        scopes=SCOPES)
 
     async def project(self) -> str:
         if self._project:
             return self._project
 
+        if _IS_DEV:
```
Comment (TheKevJames) [inline datastore/gcloud/aio/datastore/datastore.py]: I _think_ (but could be convinced either way) that this should take precedence over any `_project` value we've explicitly set, given we're effectively hijacking other user args such as `service_account` when an emulator is detected. Thoughts?

Comment (nickfirespotter) [reply datastore/gcloud/aio/datastore/datastore.py]: Worth consideration...
I think in general if a developer is going to be changing the project name between dev and prod (or staging) they have to be obtaining that project id from some variable. Either passed in to the running program or pulled from an env var. So I largely would think it wouldn't really matter.
I guess the question is if there are cases where someone would be connecting to 2 different projects in their code. And if so the emulator supports that or not... I can go either way.

Comment (TheKevJames) [reply datastore/gcloud/aio/datastore/datastore.py]: > 2 different projects
That's a great thought. We do that in a bunch of our codebases, to communicate between, say, the engineering production env and the datascience one. I suppose that would just be unsupported by the emulator. I can't really think of a decent way to support that sort of cross-project work involving an emulator, though, so I might put that off for a later version only if need be.

**Comment by TheKevJames** [comment]
Merged and rolling out as release `gcloud-aio-datastore==3.1.0`. Thanks!

**Pull request closed without merging**